Recover! method

avinoth opened the pull request:
* Add `recover!` method which raises error if the recovery fails
* Add tests and docs for `recover!`
* Add new section `callbacks` to Readme.md to document about the callbacks

NOTE: The test for `recover!` fails for Rails 5 for the same reason as #71

zzak commented:
Is this something people really want?

I'm not sure we need to add more features, but if the tests will pass I think it might be good to support both `save` and `save!` for Rails

mvz commented:
Hi @avinoth it looks like #71 was fixed. Can you please rebase this on current master?

avinoth commented:
@mvz I've rebased it with master. I've added this long time back for one of my projects, I'm not sure if it's going to be merged to the parent now? if not, feel free to close this one.

mvz commented:
Thanks, @avinoth!
